### PR TITLE
SAAS-3542: Added SQS Encryption Enabled Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/sqs/sqsEncryptionEnabled.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+
+module.exports = {
+    title: 'SQS Encryption Enabled',
+    category: 'SQS',
+    domain: 'Application Integration',
+    description: 'Ensure SQS queues are encrypted with KMS Customer Master Keys (CMKs) instead of AWS managed-keys',
```
Comment (sadeed12345) [inline plugins/aws/sqs/sqsEncryptionEnabled.js:9]:
```suggestion
    description: 'Ensure SQS queues are encrypted using keys of desired encryption level',
```

File: plugins/aws/sqs/sqsEncryptionEnabled.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+            var keyArn;
+            var kmsAliasArnMap = {};
+            listAliases.data.forEach(function(alias){
+                keyArn = alias.AliasArn.replace(/:alias\/.*/, ':key/' + alias.TargetKeyId);
+                kmsAliasArnMap[alias.AliasName] = keyArn;
+            });
+         
+            for (let queue of listQueues.data) {
+                // arn:aws:sqs:us-east-1:101363889637:aaaa
```
Comment (AkhtarAmir) [inline plugins/aws/sqs/sqsEncryptionEnabled.js:86]:
```suggestion
```

File: plugins/aws/sqs/sqsEncryptionEnabled.spec.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+];
+
+
+const getQueueAttributes = [
+    {
+        "ResponseMetadata": {
+          "RequestId": "55fecb32-ba74-528f-bea8-9174eb61d3d5"
+        },
+        "Attributes": {
+          "QueueArn": "arn:aws:sqs:us-east-1:101363889637:my-queue1",
```
Comment (AkhtarAmir) [inline plugins/aws/sqs/sqsEncryptionEnabled.spec.js:15]: Can you modify and remove account ids

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/sqs/sqsEncryptionEnabled.spec.js (hunk 3 above)
Comment (sadeed12345) [inline plugins/aws/sqs/sqsEncryptionEnabled.spec.js:15]:
```suggestion
          "QueueArn": "arn:aws:sqs:us-east-1:000111222333:my-queue1",
```

File: plugins/aws/sqs/sqsEncryptionEnabled.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+
```
Comment (sadeed12345) [inline plugins/aws/sqs/sqsEncryptionEnabled.js:4]:
```suggestion
```

File: plugins/aws/sqs/sqsEncryptionEnabled.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    description: 'Ensure SQS queues are encrypted using keys of desired encryption level',
+    more_info: 'Messages sent to SQS queues can be encrypted using KMS server-side encryption. Existing queues can be modified to add encryption with minimal overhead.'+
+        'Use customer-managed keys instead in order to gain more granular control over encryption/decryption process.',
+    recommended_action: 'Enable encryption using KMS Customer Master Keys (CMKs) for all SQS queues.',
+    link: 'http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-server-side-encryption.html',
+    apis: ['SQS:listQueues', 'SQS:getQueueAttributes', 'KMS:listAliases', 'KMS:listKeys', 
+        'KMS:describeKey', 'STS:getCallerIdentity'],
+    settings: {
+        sqs_queues_desired_encryption_level: {
+            name: 'SQS QueuesTarget Encryption Level',
```
Comment (AkhtarAmir) [inline plugins/aws/sqs/sqsEncryptionEnabled.js:17]:
```suggestion
            name: 'SQS Queues Target Encryption Level',
```

File: plugins/aws/sqs/sqsEncryptionEnabled.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+            description: 'In order (lowest to highest) sse=SSE-SQS; awskms=AWS-managed KMS; awscmk=Customer managed KMS; externalcmk=Customer managed externally sourced KMS; cloudhsm=Customer managed CloudHSM sourced KMS',
+            regex: '^(sse|awskms|awscmk|externalcmk|cloudhsm)$',
+            default: 'awscmk'
+        }
+    },
+
```
Comment (AkhtarAmir) [inline plugins/aws/sqs/sqsEncryptionEnabled.js:23]:
```suggestion
```

**Review by AkhtarAmir — APPROVED** [review]

File: plugins/aws/sqs/sqsEncryptionEnabled.spec.js [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+            sqsEncryptionEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('No SQS queues found');
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if unable to list SQS queues', function (done) {
```
Comment (dipsubha06) [inline plugins/aws/sqs/sqsEncryptionEnabled.spec.js:221]: Is it list queue error what we are checking or describe queue error

File: plugins/aws/sqs/sqsEncryptionEnabled.spec.js [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+            sqsEncryptionEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].message).to.include('Unable to query SQS queues');
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if unable to list KMS keys', function (done) {
```
Comment (dipsubha06) [inline plugins/aws/sqs/sqsEncryptionEnabled.spec.js:232]: is it list keys error or describe keys error we are checking here

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**